analyzers/tests/SonarAnalyzer.Test/TestCases/UnnecessaryBitwiseOperation.Fixed.cs

```diff
@@ -70,19 +70,19 @@ public void UnaryOperators()
 
             var bytes1 = 0;
             bytes1++;
-            _ = 0x80; // Fixed
+            _ = bytes1 | 0x80; // Compliant, See: https://github.com/SonarSource/sonar-dotnet/issues/6326
 
             var bytes2 = 0;
             ++bytes2;
-            _ = 0x80; // Fixed
+            _ = bytes2 | 0x80; // Compliant
 
             var bytes3 = 0;
             bytes3--;
-            _ = 0x80; // Fixed
+            _ = bytes3 | 0x80; // Compliant
 
             var bytes4 = 0;
             --bytes4;
-            _ = 0x80; // Fixed
+            _ = bytes4 | 0x80; // Compliant
         }
 
```

Comment:
I would add tests for all the cases of unary operators (prefix and suffix) that don't produce side effects.
For example the following.

```cs
public void UnaryOperatorsNoSideEffect()
{
    var zero = 0;
    F(+zero);
    _ = zero | 0x80; // FP: prefix unary '+' doesn't alter the value of zero

    void F(int x) { }
}
```

The list of operators to test is in the comment above on `PrefixUnaryExpressionSyntax` and `SuffixUnaryExpressionSyntax`.